### ISSUE TITLE
UPE doesn't use dynamic rendering based on allowed presentment currency

### ISSUE THREAD

**Issue by KarlisJ** [Automattic/woocommerce-payments#6517]
### Describe the bug
Affirm/Afterpay payment method is supported for USD and CAN currency but Stripe limits the currency that Affirm/Afterpay can accept payments in based on account's registered location. So a US based account can not receive payments in CAD though a CAN based account can. 

Plugin only lists both currencies for payment method resulting in both being valid options for payment method in the UI but only one currency can actually be used.

### To Reproduce

1. Checkout `add/6119-shortcode-checkout-for-bnpl` branch
2. Ensure your Stripe account is US based
3. Ensure you use multicurrency and have USD and CAD as options
4. Enable Affirm payment method (Might need [server setup](https://github.com/Automattic/woocommerce-payments-server/pull/3405))
5. Visit checkout and select USD
6. Observe Affirm as option in UPE
7. Observe that payment is successful
8. Repeat steps but select CAD as currency
9. Observe that Affirm is still available
10. Observe that payment fails

### Actual behavior
UI gets error message
>We're not able to process this request. Please refresh the page and try again.

In the logs
> ERROR The currency provided (cad) is invalid. Payments with affirm support the following currencies: usd. (invalid_request_error)

### Expected behavior
Affirm payment method should not be available as option if currency is set to CAD

**Comment by kalessil** [comment]
This needs a bit more testing:

- US account with USD, CAD deposit accounts.
- Affirm payment test for USD and CAD order
- Afterpay payment test for USD and CAD orders

If orders pass, then order currency validation shall check against merchant deposit currencies; otherwise, default merchant currency.

**Comment by brandonwanA8C** [comment]
Hey team, I'd like to understand this issue a bit more.

If the (Seller) merchant account was Onboarded as a **US entity** and configured to accept **at least both** USD and CAD,
the (Buyer) shopper initiated an Affirm Txn in Canada using CAD, the BNPL txn will fail. 
Is that the right observation?

>Affirm payment method should not be available as option if currency is set to CAD

Since Affirm [accepts both USD and CAD as the supported currency](https://stripe.com/docs/payments/buy-now-pay-later), the desired behavior is: as long as the Shopper pays with either USD or CAD, the BNPL txn with Affrim should be allowed. 
I suspect the constraint is on the Merchant Account side - can the account accept multi-currency, **regardless of the PM was used**.   

Do you/team know if the exact same behavior happens for other Payment Methods (e.g. Card txn)?

**Comment by aheckler** [comment]
I did some testing on this today. Slack thread: p1688033275712839-slack-C04VCT34PCP

TLDR of what I found, though someone should check these:

- If you're logged in, the `wcpay_selected_currency` value for your account must match up with the country of the merchant Stripe account for BNPL to work, e.g. CAD for Canadian accounts, GBP for UK accounts, and so on.
- If you're logged out and we [geolocate you](https://github.com/Automattic/woocommerce-payments/assets/5512652/d337355d-1ef9-4bd6-9410-8ed45d833a45) as inside the merchant account country, BNPL will be shown.
- If you're logged out and WC Pay geolocation is off, I'm not quite sure. It may also depend on what the default customer location is set to in core WC settings. CC @mgascam

**Comment by KarlisJ** [comment]
@brandonwanA8C 
> Do you/team know if the exact same behavior happens for other Payment Methods (e.g. Card txn)?

I'm looking into details for this issue but right away can answer to this question. No, Stripe handle BNPLs differently from other LPMs that we have used for far.

**Comment by brandonwanA8C** [comment]
@KarlisJ  
Ok, if you can try out & confirm Card and LPM txns are handled differently than BNPL, that would be great.
Let me reach out to Stripe to see if BNPL locale are handled differently by design (and why).

**Comment by brandonwanA8C** [comment]
Responded by Stripe (Shalini N.)

Basically for BNPLs like Afterpay and Affirm, payments can be accepted only with domestic currency. Eg: a US based merchant can accept Affirm/ Afterpay payments only in USD.
Sharing out [public](https://stripe.com/docs/payments/afterpay-clearpay#:~:text=local%20currency%20settlement) doc and [integration](https://docs.google.com/document/d/1ZLuVsMiAPhcEnGpAuU4OdR1EGFnfdRFgfZIk3i4Y8b4/edit) doc for reference.
I did notice that Affirm documentation is missing this explicit note...and I have passed on the feedback to the product team.

**Comment by KarlisJ** [comment]
I've created a [draft/PoC PR](https://github.com/Automattic/woocommerce-payments/pull/6708) and associated with this issue. 

It addressed the issue at checkout by removing Affirm and Afterpay from checkout if the selected currency doesn't match the account's default currency. 

However, we should investigate how this might affect the onboarding flow for merchants in the settings pages. We should look at whether we present to the merchant this difference. I believe @aheckler already did some exploration in this regard during the testing phase, maybe he can give us more insights into this.

**Comment by aheckler** [comment]
> However, we should investigate how this might affect the onboarding flow for merchants in the settings pages. We should look at whether we present to the merchant this difference.

There isn't really any onboarding flow for Affirm/Afterpay really. Merchants just check the box and that's it. I did have one suggestion in Slack though:

p1689194044876759/1689193125.877109-slack-C04VCT34PCP

**Comment by aheckler** [comment]
Did some quick and dirty testing of the feature branch, seems to work OK. :)

**Comment by KarlisJ** [comment]
@aheckler I did find one issue. 

- US based merchant account
- Remove CAD from multi-currency options (but keep multi-currency enabled)
- Try to add Affirm as LPM
![image](https://github.com/Automattic/woocommerce-payments/assets/24650518/0580c597-577c-42b0-b04b-1db2ff69e8f1)

Will look how to address this in PR.

**Comment by aheckler** [comment]
Hmm, yeah, that is odd. In that use case, customers would only be able to pay in USD anyway, right? So the fact that it's adding CAD back to the multi-currency list is a bug?

**Comment by KarlisJ** [comment]
@aheckler Yes it's a bug, I've tracked the problem in the code and looking for a solution. Will implement as part of the same associated PR.

**Comment by KarlisJ** [comment]
@aheckler Wanted to update you on decision here. We merged the fix for checkout and will address the multi-currency bug in the settings as a separate issue in #6861 . The fix for that will require larger refactor so we don't want it to hinder the customer facing issue fix.
